app/multitenant/aws_collector.go

```diff
@@ -169,14 +190,82 @@ func NewAWSCollector(config AWSCollectorConfig) (AWSCollector, error) {
 
 	// (window * report rate) * number of hosts per user * number of users
 	reportCacheSize := (int(config.Window.Seconds()) / 3) * 10 * 5
-	return &awsCollector{
+	c := &awsCollector{
 		cfg:       config,
 		db:        dynamodb.New(session.New(config.DynamoDBConfig)),
 		merger:    app.NewFastMerger(),
 		inProcess: newInProcessStore(reportCacheSize, config.Window+reportQuantisationInterval),
 		nats:      nc,
 		waiters:   map[watchKey]*nats.Subscription{},
-	}, nil
+	}
+
+	if config.StoreInterval != 0 {
+		c.ticker = time.NewTicker(config.StoreInterval)
+		go c.flushLoop()
+	}
+	return c, nil
+}
+
+func (c *awsCollector) flushLoop() {
+	for _ = range c.ticker.C {
+		c.flushPending(context.Background())
+	}
+}
+
+// Range over all users (instances) that have pending reports and send to store
+func (c *awsCollector) flushPending(ctx context.Context) {
+	instrument.CollectedRequest(ctx, "FlushPending", flushDuration, nil, func(ctx context.Context) error {
+		type queueEntry struct {
+			userid string
+			buf    []byte
+		}
+		queue := make(chan queueEntry)
+		const numParallel = 10
```

Comment:
How did you decide on the value here? Do you think it would useful to make it an arg param?

Reply:
Yes it would be more flexible, but the way flags are plumbed through from `main()` makes it annoying to add more.
There is a balancing act to get all serialisation plus IO completed within the tick interval, and I don't really know what might need adjusting.  I'm happy changing the code as I learn from running it; more than adding all possible configuration parameters then never using most of them.